app/controllers/events/import.js

```diff
@@ -0,0 +1,64 @@
+import Controller from '@ember/controller';
+import { run } from '@ember/runloop';
+
+export default Controller.extend({
+  importStatus : '',
+  importError  : '',
+  isImporting  : false,
+  file         : false,
+  fileName     : '',
+  importTask(taskUrl) {
+    run.later(() => {
+      this.get('loader')
+        .load(taskUrl)
+        .then(data => {
+          if (data.state !== 'SUCCESS') {
+            this.set('importStatus', `Status: ${data.state}`);
+            this.importTask(taskUrl);
+          } else {
+            this.set('importStatus', `Status: ${data.state}`);
+            this.transitionToRoute('events.view', data.result.id);
+          }
+        })
+        .catch(e => {
+          this.setProperties({
+            'importError' : e.message,
+            'isImporting' : false,
+            'file'        : false
+          });
+        });
+    }, 3000);
+  },
+  actions: {
+    uploadFile(files) {
+      let [file] = files;
+      var data = new FormData();
+      var endpoint = 'import/json';
+      var ext = file.name.split('.');
+      ext = ext[ext.length - 1].toLowerCase();
+      if (ext === 'xml') {
```

Comment:
What about `.zip`, `.xcal` cases ? The upload input accepts those too

Reply:
@niranjan94 actually only `.zip` files can be successfully imported, its endpoint is `import/json`. Other cases don't even have a valid endpoint, but eventyay had this code so i thought they can be implemented later on the server side. Should i only accept `.zip` then?

Reply:
@YogeshSharma0201 add the .zip and .xcal cases, it's fine if server doesn't accept them for now.